File: Solutions/S1_Q2/S1_Q2_solution.py
def Q(A, i, j, k):
	if i == j:
		return False
	else:
		if A[i] + A[j] < k:
			return Q(A, i+1, j, k)
		elif A[i] + A[j] > k:
			return Q(A, i, j-1, k)
		else:
			return True 

n = int(input())
A = list(map(int, input().split()))
k = int(input())
print(str(Q(A, 0, n-1, k)).lower())

File: Solutions/S1_Q2/S1_Q2_solution.cpp
#include <iostream>
#include <vector>
using namespace std;


bool Q(vector<int> &A, int i, int j, int k){
	
	if(i == j){
		return false;
	}
	
	if ((A[i] + A[j]) < k){
		return Q(A, i+1, j, k);
	}
	else if((A[i] + A[j]) > k){
		return Q(A, i, j-1, k);
	}
	else{
		return true;
	}
	
}

int main() {
	
	int n, k, input;
	cin >> n;
	vector<int> A;
	
	for (int i=0; i<n; i++){
		cin >> input;
		A.push_back(input);
	}
	
	cin >> k;
	bool answer = Q(A, 0, n-1, k);
	if(answer == true){
		cout << "true";
	}
	else{
		cout << "false";
	}
	return 0;
}

